docs/developer/tutorials/logs/http-crawler.md

```diff
@@ -0,0 +1,149 @@
+    """Stream of Logs from ACME"""
+```
+
+Now running *the check command* will show a new error:
+
+```
+TypeError: Can't instantiate abstract class AcmeLogStream with abstract method records
+```
+
+Once again we need to define a method, this time [`LogStream.records`](../../base/logs-crawlers.md#datadog_checks.base.checks.logs.crawler.stream.LogStream.records).
+This method accepts a `cursor` argument.
+We ignore this argument for now and explain it later.
+
+
+```python
+from datadog_checks.base.checks.logs.crawler.stream import LogRecord, LogStream
+from datadog_checks.base.utils.time import get_timestamp
+
+... # Skip AcmeCheck to focus on LogStream.
+
+
+class AcmeLogStream(LogStream):
+    """Stream of Logs from ACME"""
+
+    def records(self, cursor=None):
+        return [
+            LogRecord(
+                data={'message': 'This is a log from ACME.', 'level': 'info'},
+                cursor={'timestamp': get_timestamp()},
+            )
+        ]
+```
+
+There are several things going on here.
+`AcmeLogStream.records` returns an iterator over `LogRecord` objects.
+For simplicity here we return a list with just one record.
+After we understand what each `LogRecord` looks like we can discuss how to generate multiple records.
+
+### What is a Log Record?
+
+The `LogRecord` class has 2 fields.
+In `data` we put any data in here that we want to submit as a log to Datadog.
+In `cursor` we store a unique identifier for this specific `LogRecord`.
+
+We use the `cursor` field to checkpoint our progress as we scrape the external API.
+In other words, every time our integration completes its run we save the last cursor we submitted.
+We can then resume scraping from this cursor.
+That's what the `cursor` argument to the `records` method is for.
+The very first time the integration runs this `cursor` is `None` because we have no checkpoints.
+Every integration run after that the `cursor` will be the `LogRecord.cursor` of the last `LogRecord` that we yield or return from `records`.
+
+Some things to consider when defining cursors:
+
+- Use UTC time stamps!
+- Only using the timestamp as a unique identifier may not be enough. We can have different records with the same timestamp.
+- One popular identifier is the order of the log record in the stream. Whether this works or not depends on the API we are crawling.
+
+
+### Scraping for Log Records
+
+In our toy example we returned a list with just one record.
+In practice we will need to create list or lazy iterator over `LogRecord`s.
+We will construct them from data that we collect from the external API, in this case the one from *ACME*.
+
+Below are some tips and considerations when scraping external APIs:
+
+1. Don't forget to use the `cursor` argument to checkpoint your progress.
+1. The Agent schedules an integration run approximately every 10-15 seconds. We can treat this as a retry mechanism, so there is no need for elaborate retry or backoff logic in integrations.
```

Comment:
You might run into an API with a retry after time > 15 seconds and, if you retry every 15 seconds, it'll block you forever.

Reply:
Yeah we should remove the second sentence entirely.

Reply:
done